web/app/js/components/BaseTable.jsx

```diff
@@ -54,14 +63,37 @@ class BaseTable extends React.Component {
     this.setState({ order, orderBy });
   };
 
-  sortRows = (tableRows, tableColumns, order, orderBy) => {
-    if (!orderBy) {
+  handleFilterChange = e => {
+    let filterBy = e.target.value;
+    this.setState({ filterBy: filterBy });
+  }
+
+  generateRows = (tableRows, tableColumns, order, orderBy, filterBy) => {
+    if (!orderBy && !filterBy) {
       return tableRows;
     }
 
     let col = _find(tableColumns, d => d.dataIndex === orderBy);
     let sorted = tableRows.sort(col.sorter);
```

Comment:
i think this could error out if `orderBy === null`, though i can't reproduce it?

consider sorting only when `orderBy !== null`. also consider consolidating the return statements into a single one at the end, something like (this bit of totally untested code):
```diff
diff --git a/web/app/js/components/BaseTable.jsx b/web/app/js/components/BaseTable.jsx
index 2d9690e2..6ecfcd18 100644
--- a/web/app/js/components/BaseTable.jsx
+++ b/web/app/js/components/BaseTable.jsx
@@ -73,15 +73,16 @@ class BaseTable extends React.Component {
       return tableRows;
     }

-    let col = _find(tableColumns, d => d.dataIndex === orderBy);
-    let sorted = tableRows.sort(col.sorter);
-    if (!filterBy) {
-      return order === 'desc' ? sorted.reverse() : sorted;
-    } else {
+    let rows = tableRows;
+    if (orderBy) {
+      let col = _find(tableColumns, d => d.dataIndex === orderBy);
+      rows = rows.sort(col.sorter);
+    }
+    if (filterBy) {
       let filteredRows = [];
       let swapWildCard = /[*]/g; // replace "*"" in filterBy with wildcard
       let regexString = new RegExp(filterBy.replace(swapWildCard, ".+"), "i");
-      let extractedTableRowText = sorted.reduce((acc, row) => {
+      let extractedTableRowText = rows.reduce((acc, row) => {
         if (col.filter) {
           acc.push(col.filter(row));
         }
@@ -89,11 +90,13 @@ class BaseTable extends React.Component {
       },[]);
       extractedTableRowText.forEach((row, rowIndex) => {
         if (row.match(regexString)) {
-          filteredRows.push(sorted[rowIndex]);
+          filteredRows.push(rows[rowIndex]);
         }
       });
-      return order === 'desc' ? filteredRows.reverse() : filteredRows;
+      rows = filteredRows;
     }
+
+    return order === 'desc' ? rows.reverse() : rows;
   }

   renderHeaderCell = (col, order, orderBy) => {
```

Reply:
Took @siggy's suggestion, thank you!